source/ada/lsp-ada_handlers.adb

```diff
@@ -2114,26 +2115,147 @@ package body LSP.Ada_Handlers is
          is
             use Libadalang.Common;
 
-            Token : Token_Reference := First_Token (Node.Unit);
-            Name  : constant Wide_Wide_String :=
+            Token   : Token_Reference := First_Token (Node.Unit);
+            Name    : constant Wide_Wide_String :=
               Ada.Strings.Wide_Wide_Unbounded.To_Wide_Wide_String
                 (Get_Last_Name (Name_Node));
-            Span  : LSP.Messages.Span;
+            C       : LSP.Messages.TextEdit;
+            Span    : LSP.Messages.Span;
+            Current : Token_Reference;
+            Diff    : Integer;
+
+            Box_Line : constant Wide_String (1 .. 80) := (others => '-');
+
+            procedure Include_Result (Edit : LSP.Messages.TextEdit);
+            --  Add prepared TextEdit to Result;
+
+            function Process_Box return Boolean;
+            --  Check whether Current is box header/fother and modify it.
+            --  Return False when the searching cycle should be stopped.
+
+            --------------------
+            -- Include_Result --
+            --------------------
+
+            procedure Include_Result (Edit : LSP.Messages.TextEdit) is
+            begin
+               if not Response.result.changes (Uri).Contains (Edit) then
```

Comment:
Can we ensure unique responses by replacing algorithm itself and avoid dynamic check?

Reply:
I am open to suggestions about how it can be done.

Reply:
Why not just remove this **`if`**?

Can you provide any example where this condition is actually `True`?

Reply:
I can't guarantee that it will be always false and because of the fact that nobody except me will solve potential problems here, I would like to prevent them instead of waiting for customer's complains.

Reply:
Checking for duplicates can cost a bit if we have too many edits. Moreover, duplicates should not happen, so I am in favor of removing this 'if' since it might hide a bad behavior before this check

Reply:
Done